### ISSUE TITLE
jQuery should be removed as package dependency.

### ISSUE THREAD

**Issue by benwiley4000** [mozilla/metrics-graphics#649]
I'm building an app that uses MetricsGraphics without jQuery. Since jQuery is not required in MetricsGraphics except for tooltips, my build should not fail if I deleted the jQuery package after installing `metrics-graphics` from npm. A runtime error should be triggered if one attempts to use MetricsGraphics tooltips without jQuery available; otherwise, everything should run smoothly without.

Thoughts? I'd submit a PR, but I'm not 100% familiar with this particular build process.

**Comment by almossawi** [comment]
Correct. You should, additionally, set [show_tooltips](https://github.com/mozilla/metrics-graphics/wiki/Graphic#show_tooltips) to false.

**Comment by benwiley4000** [comment]
If I set `show_tooltips` to false then everything won't break? Interesting.. though will this avoid including jQuery in my compiled bundle?

**Comment by almossawi** [comment]
Correct. Here's the [relevant block](https://github.com/mozilla/metrics-graphics/blob/master/src/js/common/chart_title.js#L23).

**Comment by benwiley4000** [comment]
Not sure I follow. Sure that's the only *invocation* of jQuery, but doesn't this line https://github.com/mozilla/metrics-graphics/blob/master/gulp/index.js#L85 mean jQuery will be required for packaging in the bundle regardless?

**Comment by almossawi** [comment]
Ah, I misread your question. Yes, that would be a more sensible way of doing it. Self-assigning, though feel free to submit a PR in the interim if you like.

**Comment by benwiley4000** [comment]
I'll leave you to it - don't want to break anything. Thanks! :)

**Comment by benwiley4000** [comment]
I've been thinking about how this could be done and it seems like the best solution would be to set some sort of global or environment config that would determine if jQuery should be required by MG during module init. Otherwise the user would have to have specified jQuery already as a global.

**Comment by benwiley4000** [comment]
Other options: give MG a function for specifying your copy of jQuery. Or: use a non-jQuery, lighter tooltip?

**Comment by almossawi** [comment]
That seems reasonable--the referenced commit triggers an error for each chart that uses tooltips when jQuery isn't included. How does that work for you?

**Comment by benwiley4000** [comment]
Unfortunately it doesn't quite solve the problem. jQuery has been removed from the package.json but not from the gulpfile, so if I delete the jQuery module from my repo, I get this error when I bundle:

```
ERROR in ./~/metrics-graphics/dist/metricsgraphics.js
Module not found: Error: Cannot resolve module 'jquery' in /[parent directories]/node_modules/metrics-graphics/dist
 @ ./~/metrics-graphics/dist/metricsgraphics.js 3:4-37
```

Additionally - it doesn't look like the `jQuery` variable is actually defined when jQuery is provided as a module rather than a global dependency. When I tried enabling tooltips and running my app I got the error:

```
bundle.js:64335 ERROR : ... :  In order to enable tooltips, please make sure you include jQuery.
```

Although when I changed the metrics graphics code to check for `$` rather than `jQuery` I got a new error:

```
bundle.js:57568 Uncaught TypeError: $chartTitle.popover is not a function
```

Hope those error logs help.

Ben

**Comment by benwiley4000** [comment]
@almossawi does my explanation of why that fix isn't sufficient make sense?

**Comment by almossawi** [comment]
With the release planned for next Thursday, what are your thoughts on how best to proceed? Were you able to resolve your issue? From what I can tell, the change, as it stands, will be transparent to anyone upgrading. If that's not the case, or if I'm missing something crucial, feel free to point that out.
